### PR TITLE
Bump to Godwoken v1.4

### PR DESCRIPTION
**Pull request by Flouse**
# components tags
```
GODWOKEN_REF := v1.4.0-rc4 # https://github.com/nervosnetwork/godwoken/compare/v1.3.0-rc2...v1.4.0-rc4
GODWOKEN_SCRIPTS_REF := v1.3.0-rc1 # https://github.com/nervosnetwork/godwoken-scripts/compare/v1.1.0-beta...v1.3.0-rc1
```

## Other changes
- [Bump base image to ubuntu:focal](https://github.com/nervosnetwork/godwoken-docker-prebuilds/pull/71/commits/c12dcc5d0a380c6daa6857b2860ebd4255ea3498)
- [Remove ckb-indexer](https://github.com/nervosnetwork/godwoken-docker-prebuilds/pull/71/commits/858ec99971423295c392dd1b021f1640641bdbc9)
- [deprecate outdated Dockerfile](https://github.com/nervosnetwork/godwoken-docker-prebuilds/pull/71/commits/76bd8537c254f04e432012af837994c12e3e175a)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -17,7 +17,7 @@ RUN mkdir /ckb-indexer
 RUN cd /ckb-indexer && curl -LO https://github.com/nervosnetwork/ckb-indexer/releases/download/v0.3.2/ckb-indexer-0.3.2-linux.zip
 RUN cd /ckb-indexer && unzip ckb-indexer-0.3.2-linux.zip && tar xzf ckb-indexer-linux-x86_64.tar.gz
 
-FROM ubuntu:21.04
```
Comment (Flouse) [inline Dockerfile:20]: Ubuntu 21.04 is End-of-Life

File: .github/workflows/docker-publish.yml [hunk 2]
```diff
@@ -143,8 +143,8 @@ jobs:
       - name: Build godwoken
         if: steps.godwoken-cache.outputs.cache-hit != 'true'
         working-directory: build/godwoken
-        # # PORTABLE=1 USE_SSE=1 tell rocksdb to target AVX2: https://github.com/nervosnetwork/rust-rocksdb/blob/2de3ae5e5e23a315a477bd24e700eb4f5ef7a378/librocksdb-sys/build_detect_platform#L696-L699
-        run: rustup component add rustfmt && PORTABLE=1 USE_SSE=1 RUSTFLAGS="-C target-cpu=skylake" CARGO_PROFILE_RELEASE_LTO=true cargo build --release
+        # RocksDB uses -march=native. Let's use it too.
+        run: rustup component add rustfmt && RUSTFLAGS="-C target-cpu=native" CARGO_PROFILE_RELEASE_LTO=true cargo build --release
```
Comment (sopium) [inline .github/workflows/docker-publish.yml:147]: Why is this changed? Don't we still want the published images to be “mostly portable”?

Comment (Flouse) [reply .github/workflows/docker-publish.yml]: [Fix cargo build in CI for mostly portable](https://github.com/nervosnetwork/godwoken-docker-prebuilds/pull/71/commits/7fb5c9a9f6f9d82aa334e8afce9cf42da73dfb8f)

**Review by sopium — APPROVED** [review]

**Review by zeroqn — APPROVED** [review]

**Review by jjyr — APPROVED** [review]
